src/library/dao/trackdao.cpp

```diff
@@ -265,32 +264,26 @@ void TrackDAO::saveTrack(TrackPointer track) {
 }
 
 void TrackDAO::saveTrack(TrackInfoObject* pTrack) {
-    DEBUG_ASSERT_AND_HANDLE(pTrack != NULL) {
-        return;
-    }
-    //qDebug() << "TrackDAO::saveTrack" << pTrack->getId() << pTrack->getInfo();
-    TrackId trackId(pTrack->getId());
-    if (trackId.isValid()) {
-        // Track has already been stored in database
-        if (pTrack->isDirty()) {
-            //qDebug() << this << "Dirty tracks before clean save:" << m_dirtyTracks.size();
-            //qDebug() << "TrackDAO::saveTrack. Dirty. Calling update";
-            updateTrack(pTrack);
+    DEBUG_ASSERT(nullptr != pTrack);
```

Comment:
BTW I would prefer to keep "_AND_HANDLE" for things that would easily prevent a crash.

It may be impossible today but you never know what tomorrow will bring :).